File: code-examples/php/usage-examples/insertOne.php
<?php
require_once __DIR__ . "/vendor/autoload.php";

class Haiku {
  public $title;
  public $content;

  function __construct($title, $content){
    $this->title = $title;
    $this->content = $content;
  }
}

$uri = "mongodb://localhost:27017";
$client = new MongoDB\Client($uri);

$collection = $client->poetry->haiku;

try {
    $newHaiku = new Haiku(
        "Record of a Shriveled Datum", 
        "No bytes, no problem. Just insert a document, in MongoDB"
    );

    $result = $collection->insertOne($newHaiku);

    print("A document was inserted with the _id:" . $result->getInsertedId());
} catch(Exception $e){
    echo $e->getMessage();
}
?>

File: code-examples/scala/usage-examples/insertOne.scala
import org.mongodb.scala._

// Replace the uri string with your MongoDB deployment's connection string.
val uri = "<connection string uri>"

val mongoClient: MongoClient = MongoClient(uri)

val database: MongoDatabase = mongoClient.getDatabase("insertDB")

val haiku: MongoCollection[Document] = database.getCollection("haiku")

val doc: Document = Document("title" -> "Record of a Shriveled Datum",
      "content" -> "No bytes, no problem. Just insert a document, in MongoDB")

haiku.insertOne(doc).printHeadResult("Insert Result: ")
    , (results: InsertOneResult) => println("A document was inserted with the _id: " + results.getInsertedId))

mongoClient.close().printHeadResult()
